core/types/calls.cc

```diff
@@ -3648,6 +3648,130 @@ class Magic_mergeHashValues : public IntrinsicMethod {
     }
 } Magic_mergeHashValues;
 
+namespace {
+
+void digImplementation(const GlobalState &gs, const DispatchArgs &args, DispatchResult &res, NameRef methodToDigWith) {
+    if (args.args.size() == 0 || args.numPosArgs != args.args.size()) {
+        // A type error was already reported for arg mismatch
+        return;
+    }
+
+    auto baseCaseArgLocs = InlinedVector<LocOffsets, 2>{};
+    baseCaseArgLocs.emplace_back(args.locs.args[0]);
+    auto baseCaseLocs = CallLocs{
+        args.locs.file,                         // file
+        args.locs.call.copyWithZeroLength(),    // call
+        args.locs.args[0].copyWithZeroLength(), // receiver
+        args.locs.args[0].copyWithZeroLength(), // fun
```

Comment:
I don't think so?

```ruby
opts.dig("first", "second")
     ^^^ fun
         ^^^^^^^ args[0]
```

Having it be zero length makes it basically ignore it for most things, and use the `callLoc` instead, except that it's very hard to get a case where it would need to use these errors, because we know that the `Array#at` / `Hash#[]` methods always exist, so we're unlikely to need the `fun` locs. The most important location in practice is the basecaseArgLocs in case you try to pass `0` to a `T::Hash[String, ...]` or something.